internal/services/synapse/synapse_pipeline_resource.go

```diff
@@ -0,0 +1,361 @@
+				ForceNew:     true,
+				ValidateFunc: validate.WorkspaceID,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"variables": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"activities_json": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: suppressJsonOrderingDifference,
+			},
+
+			"annotations": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceSynapsePipelineCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	synapseClient := meta.(*clients.Client).Synapse
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	environment := meta.(*clients.Client).Account.Environment
+
+	workspaceId, err := parse.WorkspaceID(d.Get("synapse_workspace_id").(string))
+	if err != nil {
+		return err
+	}
+
+	client, err := synapseClient.PipelinesClient(workspaceId.Name, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] preparing arguments for Synapse Pipeline creation.")
+
+	id := parse.NewPipelineID(workspaceId.SubscriptionId, workspaceId.ResourceGroup, workspaceId.Name, d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.GetPipeline(ctx, id.Name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing %s: %+v", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_synapse_pipeline", id.ID())
+		}
+	}
+
+	pipeline := &artifacts.Pipeline{
+		Parameters:  expandSynapseParameters(d.Get("parameters").(map[string]interface{})),
+		Variables:   expandSynapseVariables(d.Get("variables").(map[string]interface{})),
+		Description: utils.String(d.Get("description").(string)),
+	}
+
+	if v, ok := d.GetOk("activities_json"); ok {
+		activities, err := deserializeSynapsePipelineActivities(v.(string))
+		if err != nil {
+			return fmt.Errorf("parsing 'activities_json' for %s", err)
+		}
+		pipeline.Activities = activities
+	}
+
+	if v, ok := d.GetOk("annotations"); ok {
+		annotations := v.([]interface{})
+		pipeline.Annotations = &annotations
+	} else {
+		annotations := make([]interface{}, 0)
+		pipeline.Annotations = &annotations
+	}
+
+	config := artifacts.PipelineResource{
+		Pipeline: pipeline,
+	}
+
+	future, err := client.CreateOrUpdatePipeline(ctx, id.Name, config, "")
+	if err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
```

Comment:
Minor issue, but would be more Go idiomatic:

```suggestion
	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
```

Reply:
Fixed